patches/tModLoader/Terraria/GameContent/UI/ResourceSets/ClassicPlayerResourcesDisplaySet.cs.patch

```diff
@@ -0,0 +1,226 @@
++				
+ 				int a = (int)((double)num2 * 0.9);
++
++				double segmentResourceMax = (i + 1) * UIDisplay_ManaPerStar;
++				double segmentResourceMaxFromCurrent = ((snapshot.Mana / UIDisplay_ManaPerStar) + 1) * UIDisplay_ManaPerStar;
++				double resource = segmentResourceMax > segmentResourceMaxFromCurrent ? -1 : Utils.Clamp(segmentResourceMax, 0, snapshot.Mana);
++				ResourceOverlayDrawContext drawContext = new ResourceOverlayDrawContext(snapshot, this, resource, i, TextureAssets.Mana) {
++					position = position,
++					color = new Color(num2, num2, num2, a),
++					origin = TextureAssets.Mana.Size() / 2f,
++					scale = new Vector2(num3)
++				};
++
++				ResourceOverlayLoader.DrawResource(drawContext);
++				/*
+ 				spriteBatch.Draw(TextureAssets.Mana.Value, new Vector2(775 + UI_ScreenAnchorX, (float)(30 + TextureAssets.Mana.Height() / 2) + ((float)TextureAssets.Mana.Height() - (float)TextureAssets.Mana.Height() * num3) / 2f + (float)(28 * (i - 1))), new Rectangle(0, 0, TextureAssets.Mana.Width(), TextureAssets.Mana.Height()), new Color(num2, num2, num2, a), 0f, new Vector2(TextureAssets.Mana.Width() / 2, TextureAssets.Mana.Height() / 2), num3, SpriteEffects.None, 0f);
++				*/
+ 			}
++
++			SkipStars:
++			ResourceOverlayLoader.PostDrawResourceDisplay(snapshot, this, false, color, drawText);
+ 		}
+ 
+ 		public void TryToHover() {
+ 			Vector2 mouseScreen = Main.MouseScreen;
+ 			Player localPlayer = Main.LocalPlayer;
+-			int num = 26 * localPlayer.statLifeMax2 / (int)UIDisplay_LifePerHeart;
+-			int num2 = 0;
+-			if (localPlayer.statLifeMax2 > 200) {
++
++			PlayerStatsSnapshot snapshot = new PlayerStatsSnapshot(localPlayer);
++
++			int num = 26 * snapshot.AmountOfLifeHearts;
++			float num2 = 0;
++			if (snapshot.AmountOfLifeHearts > 10) {
+ 				num = 260;
+ 				num2 += 26;
+ 			}
+ 
+-			if (mouseScreen.X > (float)(500 + UI_ScreenAnchorX) && mouseScreen.X < (float)(500 + num + UI_ScreenAnchorX) && mouseScreen.Y > 32f && mouseScreen.Y < (float)(32 + TextureAssets.Heart.Height() + num2))
++			if (mouseScreen.X > (float)(500 + UI_ScreenAnchorX) && mouseScreen.X < (float)(500 + num + UI_ScreenAnchorX) && mouseScreen.Y > 32f && mouseScreen.Y < (float)(32 + TextureAssets.Heart.Height() + num2) && ResourceOverlayLoader.DisplayHoverText(snapshot, this, true))
+ 				CommonResourceBarMethods.DrawLifeMouseOver();
+ 
+ 			num = 24;
+-			num2 = 28 * localPlayer.statManaMax2 / UIDisplay_ManaPerStar;
+-			if (mouseScreen.X > (float)(762 + UI_ScreenAnchorX) && mouseScreen.X < (float)(762 + num + UI_ScreenAnchorX) && mouseScreen.Y > 30f && mouseScreen.Y < (float)(30 + num2))
++			num2 = 28 * snapshot.AmountOfManaStars;
++			if (mouseScreen.X > (float)(762 + UI_ScreenAnchorX) && mouseScreen.X < (float)(762 + num + UI_ScreenAnchorX) && mouseScreen.Y > 30f && mouseScreen.Y < (float)(30 + num2) && ResourceOverlayLoader.DisplayHoverText(snapshot, this, false))
+ 				CommonResourceBarMethods.DrawManaMouseOver();
```

Comment:
Is there any way we can reduce the patch size of this file? There's so much here that could break in an update

Reply:
I guess you don't need to use `snapshot` everywhere now that the hook is removed?

Reply:
yeah, i could safely revert the `snapshot` changes in the Classic display set due to the hook not being present anymore